src/cliffordsynthesis/CliffordSynthesizer.cpp

```diff
@@ -395,4 +416,82 @@ void CliffordSynthesizer::updateResults(const Configuration& config,
     break;
   }
 }
+
+// assume canonical sorting of gates
+std::vector<std::size_t> getLayers(const qc::QuantumComputation& qc) {
+  std::vector<std::size_t> layerNum{};
+  layerNum.resize(qc.size());
+  std::vector<std::size_t> layers{};
+  std::size_t              layer = 0U;
+  for (std::size_t i = 0; i < qc.size(); ++i) {
+    const auto& gate = *(qc.begin() + i);
+
+    for (const auto& qubit : gate->getUsedQubits()) {
+      if (layerNum[qubit] >= layer) {
+        ++layer;
+        layers.emplace_back(i);
+        break;
+      }
+    }
+    for (const auto& qubit : gate->getUsedQubits()) {
+      layerNum[qubit] = layer;
+    }
+  }
+  if (layers.back() < qc.size()) {
+    layers.emplace_back(qc.size());
+  }
+  return layers;
+}
+
+void CliffordSynthesizer::depthHeuristicSynthesis() {
+  INFO() << "Optimizing Circuit with Heuristic" << std::endl;
+  auto optimalConfig                 = configuration;
+  optimalConfig.heuristic            = false;
+  optimalConfig.target               = TargetMetric::Depth;
+  optimalConfig.initialTimestepLimit = configuration.splitSize;
+
+  qc::CircuitOptimizer::reorderOperations(*initialCircuit);
+  qc::QuantumComputation   optCircuit{initialCircuit->getNqubits()};
+  std::vector<std::size_t> layers = getLayers(*initialCircuit);
```

Comment:
Maybe I have overlooked something, but I believe this can be `const`

Reply:
Weird that not even SonarLint suggested that.

Reply:
That's what the `const` police (aka me) is for 😜